### PR TITLE
Leetcode Scrapper

### LINKED ISSUE #753: LeetCode Scrapper
# Details

Come up with a script that scraps `n` number of problems from `any category/difficulty` provided by the user and store all the `n` problems separately in PDFs. The PDF file should contain the problem title, statement, and test cases.

User Input: `n` and `category/difficulty` of problem
Expected Output: `n` PDF files

### PR DESCRIPTION
**Pull request by AshuKV**
# Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

## Fixes #753 

Replace `issue_no` with the issue number which is fixed in this PR

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

_Please delete options that are not relevant._

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [x] Documentation Update

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

File: LeetCode-Scrapper/ques.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+from fpdf import FPDF
+
+options = webdriver.ChromeOptions()
+options.add_argument("--headless")
+
+
+capa = DesiredCapabilities.CHROME
+capa["pageLoadStrategy"] = "none"
+
+print("Enter Chrome Driver path")
```
Comment (kaustubhgupta) [inline LeetCode-Scrapper/ques.py:18]:
```suggestion
print("Enter Chrome Driver path: ")
```

File: LeetCode-Scrapper/ques.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+inp = input()
```
Comment (kaustubhgupta) [inline LeetCode-Scrapper/ques.py:19]: use better variable name

File: LeetCode-Scrapper/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# LeetCode Scraper
+This python script will let the user to scrape 'n' number of LeetCode problems from any category/difficulty in [Leetcode](https://leetcode.com/problemset/all), as provided by the user. The functionality of the script is to gain the information regarding particular leetcode problem in different PDFs.
+
+## Prerequisites:
+Download the required packages from the following command in you terminal.(Make sure you're in the same project directory)
+
+` pip3 install -r requirements.txt `
+
+To run this script, you need to have selenium installed and configure webdriver to use chrome browser in your$PATH. You can directly download chrome driver from the link below- https://chromedriver.chromium.org/downloads. Further, if this argument is not specified, it will search path, otherwise you can set the path to chromedriver using
```
Comment (kaustubhgupta) [inline LeetCode-Scrapper/README.md:9]: Do you think this chrome driver explanation is still needed when you have taken input for the chrome driver path?

File: LeetCode-Scrapper/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+` driver = webdriver.Chrome('/path/to/chromedriver') `
```
Comment (kaustubhgupta) [inline LeetCode-Scrapper/README.md:11]: not needed

File: LeetCode-Scrapper/ques.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    test_case=problem["test_case"].encode('latin-1', 'replace').decode('latin-1')
+    #set url
+    url=problem["url"]
+    pdf.cell(200, 10, txt =title, ln = 1, align = 'C')
+    pdf.multi_cell(200, 10, txt =statement, align = 'L')
+    pdf.multi_cell(200, 10, txt =test_case, align = 'L')
+    pdf.write(5, 'Problem_Link: ')
+    pdf.write(5,url,url)
+    title = title.rstrip()
+    pdf.output(title+".pdf")
```
Comment (kaustubhgupta) [inline LeetCode-Scrapper/ques.py:93]:
```suggestion
    pdf.output("./LeetCode-Scrapper/"+ title+".pdf")   
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by kaustubhgupta** [comment]
@AshuKV updates on this?

**Comment by kaustubhgupta** [comment]
@AshuKV will you update the PR today? that's the max extension I can give now, else the PR will be closed. Then don't message us that you were not informed.

**Review by AshuKV — COMMENTED** [review]
I have resolved the changes. You could merge the PR now.

**Review by kaustubhgupta — APPROVED** [review]

**Review by antrikshmisri — APPROVED** [review]

**Review by santushtisharma10 — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]
LGTM

**Pull request merged**